sphinx_gallery/tests/test_py_source_parser.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+r"""
+Test source parser
+==================
+
+
+"""
+# Author: Óscar Nájera
+# License: 3-clause BSD
+
+from __future__ import division, absolute_import, print_function
+import sphinx_gallery.py_source_parser as sg
+
+
+def test_get_docstring_and_rest():
+
+    docstring, rest, lineno = sg.get_docstring_and_rest(
+        'sphinx_gallery/tests/unicode.sample')
```

Comment:
Can you write the content into a temporary file instead using `tmpdir`? This way the test is a bit easier to understand.

Reply:
Not really. If you check on the previous file there is `test_identify_names2` there I indeed write to file. And for everything to work correctly in py2 & py3 I write it as bytes. But in this case and the function `test_identify_names` I really want to read a file from disk, the file is identical. But is the 2 places I need to be sure the file read from disk is correctly parsed.

Reply:
It seems like this does work:

```py

content = b'\n'.join(
    [b'',
     b'# -*- coding: utf-8 -*-',
     b"'''",
     b'\xc3\x9anicode in header',
     b'=================',
     b'',
     b'U\xc3\xb1icode in description',
     b"'''",
     b'',
     b'# Code source: \xc3\x93scar N\xc3\xa1jera',
     b'# License: BSD 3 clause',
     b'',
     b'import os',
     b"path = os.path.join('a','b')",
     b'',
     b"a = 'hei\xc3\x9f'  # Unicode string",
     b'',
     b'import sphinx_gallery.back_references as br',
     b'br.identify_names',
     b'',
     b'from sphinx_gallery.back_references import identify_names',
     b'identify_names ',
     b''])

open('/tmp/unicode.sample', 'wb').write(content)

# use tmpdir to get a file here instead
open('/tmp/unicode.sample', 'wb').write(content)
```

Reply:
Are you sure this is simpler than having a unicode heavy python file on disk? Also, I use this file twice, one to test that despite the unicode I `get_docstring_and_rest` and one to test that despite the unicode I `identify_names` of the used functions.

Reply:
> Are you sure this is simpler than having a unicode heavy python file on disk?

Yes, this is a lot more clear because you see exactly what you are writing to the file. Otherwise how do we know how you generated your file. Also test files like this don't get included in the installed package which mean that you can not run the tests on an installed package to make sure that they run. For example this is a problem we already have for sphinx_gallery/tests/reference_parse.txt.

If you use the file content in two tests you can put that in a helper function.

Reply:
OK then. I implemented a pytest fixture to generate this sample file